### ISSUE TITLE
[Feature Request] More chroots support in one config

### ISSUE THREAD

**Issue by paulcarroty** [nrechn/mock-travis#5]
It will be great if maintainer can do something like

``` yaml
mock_config:
 - fedora-24-x86_64
 - fedora-25-x86_64
 - $another_distroname
```

**Comment by nrechn** [comment]
Thanks for providing idea. 

```yml
mock_config:
 - fedora-24-x86_64
 - fedora-25-x86_64
```

Does it mean test packaging on both Fedora 24 and Fedora 25 (or even more Fedora/CentOS version) in one Travis-CI build test?

**Comment by paulcarroty** [comment]
@nrechn yes. Because create new commit for every OS is overhead.

**Comment by kuboosoft** [comment]
taking advantage of the image, Debian or Fedora?  debian provides a old version of "mock" (1.1.33/1.2.18), and fedora 24  (1.2.21); exist a way to add a target.cfg to '/etc/mock/' manually? @nrechn 
maybe a "sudo echo 'foo configure'......" 
or 
'wget -c -P /etc/mock/ https://foo/fedora-24-x86_64.cfg'

**Comment by paulcarroty** [comment]
@kuboosoft there is F24 in docker image. I created the [F26 image](https://hub.docker.com/r/paulcarroty/fedora-mock/) if you need it.

About custom config - IMHO this is overengineering, much more easy is container autoupdate every 3-6 months.

**Comment by nrechn** [comment]
Hi @paulcarroty ,

Sorry for late response. I was busy on my uni final exams of this semester.

I updated `mock-travis`. It supports multiple mock configs now :tada: .

In `mock_travis` part of `.travis.yml`, it is able to set multiple mock configs. For example:
``` yml
mock_travis:
  mock_config:
    - fedora-25-x86_64
    - fedora-24-x86_64
```

I have done a simple test on this feature. Here is the [travis-ci log of my test](https://travis-ci.org/nrechn/Sway-Fedora/builds/179150454).

Hope it helps.

**Comment by paulcarroty** [comment]
Hi @nrechn, it's great, good luck in your education.

Thanks for multiple mock configs, good feature. One sad fact - in Travis we have the 40 min time limits, and 2-3 mock configs maybe will not enough for compilation of middle/big projects. Anyway, the feature maybe will be good for another CI.

I created [the fork](https://github.com/paulcarroty/mock-travis) with enabled verbose mode and no clean cache for mock (`-v --no-clean`). I also disabled delete a docker image after build because it's only time lost; enable deploy result packages to GitHub releases.

[Here](https://github.com/UnitedRPMs/travis-rpm-builder) we build packages for UnitedRPMs.

If you're interested, I can create the pull-request with this changes.

**Comment by paulcarroty** [comment]
Note: verbose mode can help to win a 10 min no-output logs of Travis timeout.

**Comment by nrechn** [comment]
Hi @paulcarroty 

Thanks your advice. I agree `delete a docker image` is time lost. I removed this process in commit c07da63966dff876cbf29f5404dade2639ee653a.

<br>

> Note: verbose mode can help to win a 10 min no-output logs of Travis timeout.

Do you mean "If no output is received from a build for 10 minutes, it’s assumed to have stalled for unknown reasons and is subsequently killed" ?

I set mock-travis to output less. Otherwise, output would run out of maximum Travis log size. If you wish, I could add a setting of verbose mode in order to allow users choose whether to enable verbose mode.
e.g. `verbose_mode: enable`

**Comment by paulcarroty** [comment]
> Do you mean "If no output is received from a build for 10 minutes, it’s assumed to have stalled for unknown reasons and is subsequently killed" ?

Definitely, I get it with atom&apm builds. It have A LOT of JS/npm warnings.

> If you wish, I could add a setting of verbose mode in order to allow users choose whether to enable verbose mode. e.g. verbose_mode: enable

Great, do it. I use two builds (binaries) for verbose/non-verbose mode. Not the best way, know it.

Another problem with the log limit (AFAIR 4 Mb): without logs no one can't resolve any problem, but if we need to build projects with big logs - small timer with disabled logs can help. For example: `Start building.. 5m... 10m... 15m...` etc.

Anyway, I need more IRL test and can tell more.

**Comment by kuboosoft** [comment]
Generally the real problem is the limit time, and run docker as admin... In this moment only exist 2 ci services without problem... the powerful ci services not permit it...
